test/mpl/integ/test_required_ec_cmm.py

```diff
@@ -0,0 +1,737 @@
+SOME_MANY_ITEM_ENCRYPTION_CONTEXT = {
+    ''.join(random.choices(string.ascii_letters, k=6))
+    : ''.join(random.choices(string.ascii_letters, k=6)) for _ in range(20)
+}
+ENCRYPTION_CONTEXT_SUITE = [
+    SOME_EMPTY_ENCRYPTION_CONTEXT,
+    SOME_SINGLE_ITEM_ENCRYPTION_CONTEXT,
+    SOME_DOUBLE_ITEM_ENCRYPTION_CONTEXT,
+    SOME_MANY_ITEM_ENCRYPTION_CONTEXT,
+]
+
+# Keyrings under test
+SOME_RSA_KEYRING = _create_raw_rsa_keyring()
+SOME_AES_KEYRING = _create_raw_aes_keyring()
+SOME_KMS_KEYRING = _create_kms_keyring()
+SOME_HIERARCHICAL_KEYRING = _create_hierarchical_keyring()
+TEST_KEYRINGS_LIST = [
+    SOME_AES_KEYRING,
+    SOME_KMS_KEYRING,
+    SOME_RSA_KEYRING,
+    SOME_HIERARCHICAL_KEYRING,
+]
+# Multi-keyring composed of individual keyrings.
+# In lots of tests in this file,
+# this multi keyring encrypts one message,
+# then the test attempts decryption with all of its component keyrings
+# ("component" = generator + child keyrings).
+SOME_MULTI_KEYRING = _create_multi_keyring(TEST_KEYRINGS_LIST)
+
+
+SOME_PLAINTEXT = b"Hello World"
+
+
+@pytest.mark.parametrize("encryption_context", ENCRYPTION_CONTEXT_SUITE)
+# HAPPY CASE 1
+# Test supply same encryption context on encrypt and decrypt NO filtering
+def test_GIVEN_same_EC_on_encrypt_and_decrypt_WHEN_encrypt_decrypt_cycle_THEN_decrypt_matches_plaintext(
+    encryption_context
+):
+    # When: encrypt/decrypt cycle
+    ct, _ = client.encrypt(
+        source=SOME_PLAINTEXT,
+        keyring=SOME_MULTI_KEYRING,
+        # Given: same encryption context on encrypt and decrypt
+        encryption_context=encryption_context,
+    )
+
+    for decrypt_keyring in TEST_KEYRINGS_LIST:
+        pt, _ = client.decrypt(
+            source=ct,
+            # Given: same encryption context on encrypt and decrypt
+            encryption_context=encryption_context,
+            keyring=decrypt_keyring
+        )
+
+        # Then: decrypted plaintext matches original plaintext
+        assert pt == SOME_PLAINTEXT
+
+
+@pytest.mark.parametrize("encryption_context", ENCRYPTION_CONTEXT_SUITE)
+# HAPPY CASE 2
+# On Encrypt we will only write one encryption context key value to the header
+# we will then supply only what we didn't write wth no required ec cmm,
+# This test case is checking that the default cmm is doing the correct filtering
+def test_GIVEN_RECCMM_with_one_REC_key_on_encrypt_AND_default_CMM_with_valid_reproduced_EC_on_decrypt_WHEN_supply_reproduced_EC_with_REC_key_on_decrypt_THEN_decrypt_matches_plaintext( # noqa pylint: disable=line-too-long
+    encryption_context
+):
+    # This test needs >1 item to supply as required encryption context
+    if len(encryption_context) < 1:
+        return
+
+    # Grab one item from encryption_context to supply as reproduced EC
+    one_k, one_v = next(iter(encryption_context.items()))
+    reproduced_ec = {one_k: one_v}
+    # Given: one required encryption context (REC) key
```

Comment:
I love these comments. Is it possible for you to use `duvet` to align these with https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/test-vectors/complete-vectors/required-encryption-context-cmm.md?

Reply:
Chatted and decided this isn't worth it.